src/templates/TemplateData.ts

```diff
@@ -133,107 +133,35 @@ function verifyTemplatesByRuntime(templatesMap: { [runtime: string]: Template[]
-                const runtime: ProjectRuntime = <ProjectRuntime>ProjectRuntime[key];
-                const feedRuntime: string = getFeedRuntime(runtime);
-                const releaseVersion: string = userTemplateVersion ? userTemplateVersion : cliFeedJson.tags[feedRuntime].release;
-                const cachedRelease: string | undefined = globalState.get(`${runtime}-release`);
-                if (!cachedRelease || releaseVersion !== cachedRelease) {
-                    // templates are not up-to-date and need to be downloaded/extracted
-                    return undefined;
-                }
-
-                const cachedResources: object | undefined = globalState.get<object>(getRuntimeKey(resourcesKey, runtime));
-                const cachedTemplates: object[] | undefined = globalState.get<object[]>(getRuntimeKey(templatesKey, runtime));
-                const cachedConfig: object | undefined = globalState.get<object>(getRuntimeKey(configKey, runtime));
-
-                if (cachedResources && cachedTemplates && cachedConfig) {
-                    [templatesMap[runtime], configMap[runtime]] = parseTemplates(cachedResources, cachedTemplates, cachedConfig);
-                    verifyTemplatesByRuntime(templatesMap, runtime);
-                } else {
-                    return undefined;
-                }
-            }
-
-            return new TemplateData(templatesMap, configMap);
-        });
-    } catch (error) {
-        return undefined;
-    }
-}
-
-export async function tryGetLatestTemplateData(reporter: TelemetryReporter | undefined, cliFeedJson: cliFeedJsonResponse, globalState?: vscode.Memento): Promise<TemplateData | undefined> {
+export async function tryGetTemplateData(reporter: TelemetryReporter | undefined, cliFeedJson: cliFeedJsonResponse, globalState?: vscode.Memento, isBackup: boolean = false): Promise<TemplateData | undefined> {
+    const templateReleaseVersion: string = 'templateReleaseVersion';
+    const v1ReleaseVersion: string = '1.0.12'; // known stable version
+    const betaReleaseVersion: string = '2.0.1-beta.25'; // known stable version
     try {
         return <TemplateData>await callWithTelemetryAndErrorHandling('azureFunctions.tryGetLatestTemplateData', reporter, undefined, async function (this: IActionContext): Promise<TemplateData> {
             this.suppressErrorDisplay = true;
             this.properties.isActivationEvent = 'true';
             const templatesMap: { [runtime: string]: Template[] } = {};
             const configMap: { [runtime: string]: Config } = {};
             for (const key of Object.keys(ProjectRuntime)) {
+                const runtime: ProjectRuntime = <ProjectRuntime>ProjectRuntime[key];
                 // called within loop in case setting has changed between runtimes
                 const userTemplateVersion: string | undefined = getFuncExtensionSetting(templateVersionSetting);
```

Comment:
I'm still not happy with `invalidTemplateVersionPrompt` and the fact that you're getting the user setting multiple times. How about you rename `invalidTemplateVersionPrompt` to `getTemplateVersionSetting` and call it once. It gets the setting and validates it at the same time, _before_ you enter the loop or try to get any templates